pangeo_forge_recipes/transforms.py

```diff
@@ -383,20 +383,81 @@ class StoreToZarr(beam.PTransform, ZarrWriterMixin):
     :param combine_dims: The dimensions to combine
     :param target_chunks: Dictionary mapping dimension names to chunks sizes.
         If a dimension is a not named, the chunks will be inferred from the data.
+    :param target_chunk_aspect_ratio: Dictionary mapping dimension names to desired
+        aspect ratio of total number of chunks along each dimension.
+    :param target_chunk_size: Desired single chunks size. Can be provided as
+        integer (bytes) or as a str like '100MB' etc.
+    :param size_tolerance : float, optional
+        Chunksize tolerance. Resulting chunk size will be within
+        [target_chunk_size*(1-size_tolerance),
+        target_chunk_size*(1+size_tolerance)] , by default 0.2
     """
 
     # TODO: make it so we don't have to explicitly specify combine_dims
     # Could be inferred from the pattern instead
     combine_dims: List[Dimension]
-    target_chunks: Dict[str, int] = field(default_factory=dict)
+    target_chunks: Optional[Dict[str, int]] = field(default=None)
+    target_chunks_aspect_ratio: Optional[Dict[str, int]] = field(default=None)
+    target_chunk_size: Optional[Union[str, int]] = field(
+        default=None
+    )  # ? Should we provide a default?
+    size_tolerance: float = 0.2
+
+    def __post_init__(self):
+        # if none of the chunking parameters is specified, set the default behavior
+        # of target_chunks={}
+        if all(
+            a is None
+            for a in (self.target_chunks, self.target_chunk_size, self.target_chunks_aspect_ratio)
+        ):
+            self.target_chunks = {}
+
+        # check that not both static and dynamic chunking are specified
+        if self.target_chunks is not None and (
+            self.target_chunk_size is not None or self.target_chunks_aspect_ratio is not None
+        ):
+            raise ValueError(
+                (
+                    "Cannot specify both target_chunks and "
+                    "target_chunk_size or target_chunks_aspect_ratio."
+                )
+            )
+
+        # if dynamic chunking is specified, make sure both target_chunk_size
+        # and target_chunks_aspect_ratio are specified
+        if self.target_chunks is None and not (
+            self.target_chunk_size is not None and self.target_chunks_aspect_ratio is not None
+        ):
+            raise ValueError(
+                (
+                    "Must specify both target_chunk_size and "
+                    "target_chunks_aspect_ratio to enable dynamic chunking."
+                )
+            )
```

Comment:
```suggestion
        if self.target_chunks is None and not (
            self.target_chunk_size and self.target_chunks_aspect_ratio
        ):
            raise ValueError(
                "To enable dynamic chunking, both `target_chunk_size` and "
                "`target_chunks_aspect_ratio` must be specified."
            )
```

- I think phrased this way, the ValueError message can stand in for the comment. 
- Also I think restating the conditional expression without `is not None` will work (same as above) and is more succinct.
- The extra layer of parentheses on the error message should not be necessary.